tests/permissions/base_permissions_framework_test.py

```diff
@@ -0,0 +1,109 @@
+"""
+ARCHES - a program developed to inventory and manage immovable cultural heritage.
+Copyright (C) 2013 J. Paul Getty Trust and World Monuments Fund
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from tests.base_test import ArchesTestCase
+
+import os
+from tests import test_settings
+from tests.base_test import ArchesTestCase
+from django.core import management
+from django.urls import reverse
+from django.test.client import RequestFactory, Client
+from django.test.utils import captured_stdout
+from arches.app.utils.betterJSONSerializer import JSONSerializer, JSONDeserializer
+from django.contrib.auth.models import User
+from django.contrib.auth.models import Group
+from arches.app.models.models import GraphModel, ResourceInstance, Node
+from arches.app.models.resource import Resource
+from arches.app.permissions.arches_default_deny import (
+    ArchesDefaultDenyPermissionFramework,
+)
+
+
+class ArchesPermissionFrameworkTestCase(ArchesTestCase):
+    def setUp(self):
```

Comment:
Nothing fails if I do this to allow the setup to run 1x per class instead of per test...
```suggestion
    @classmethod
    def setUpTestData(cls):
        cls.add_users()
        # change self to cls below
```
...